h2o-algos/src/main/java/hex/modelselection/ModelSelection.java

```diff
@@ -0,0 +1,334 @@
+    }
+    
+    public ModelSelection(hex.modelselection.ModelSelectionModel.ModelSelectionParameters parms) {
+        super(parms);
+        init(false);
+    }
+    
+    public ModelSelection(hex.modelselection.ModelSelectionModel.ModelSelectionParameters parms, Key<hex.modelselection.ModelSelectionModel> key) {
+        super(parms, key);
+        init(false);
+    }
+    
+    @Override
+    protected int nModelsInParallel(int folds) {
+        return nModelsInParallel(1,2);  // disallow nfold cross-validation
+    }
+
+    @Override
+    protected ModelSelectionDriver trainModelImpl() {
+        return new ModelSelectionDriver();
+    }
+
+    @Override
+    public ModelCategory[] can_build() {
+        return new ModelCategory[]{ModelCategory.Regression};
+    } // because of r2 usage
+
+    @Override
+    public boolean isSupervised() {
+        return true;
+    }
+
+    @Override
+    public boolean haveMojo() {
+        return false;
+    }
+
+    @Override
+    public boolean havePojo() {
+        return false;
+    }
+
+    public ModelBuilder.BuilderVisibility buildVisibility() {
+        return ModelBuilder.BuilderVisibility.Experimental;
+    }
+
+    public void init(boolean expensive) {
+        if (_parms._nfolds > 0 || _parms._fold_column != null) {
+            _glmNFolds = _parms._nfolds;
+            if (_parms._fold_assignment != null) {
+                _foldAssignment = _parms._fold_assignment;
+                _parms._fold_assignment = null;
+            }
+            if (_parms._fold_column != null) {
+                _foldColumn = _parms._fold_column;
+                _parms._fold_column = null;
+            }
+            _parms._nfolds = 0;
+        }
+        super.init(expensive);
+        if (expensive) {
+            initModelSelectionParameters();
+            initModelParameters();
+        }
+    }
+    
+    private void initModelParameters() {
+        _numModelBuilt = calculateModelNumber(_numPredictors, _parms._max_predictor_number);
+        _bestR2Values = new double[_parms._max_predictor_number];
+        _bestModelPredictors = new String[_parms._max_predictor_number][];
+    }
+
+    private void initModelSelectionParameters() {
+        if (nclasses() > 1)
+            error("response", "ModelSelection only works with regression.");
+        
+        if (!(AUTO.equals(_parms._family) || gaussian.equals(_parms._family)))
+            error("_family", "ModelSelection only supports Gaussian family");
+        
+        if (_parms._nparallelism < 0) 
+            error("nparallelism", "must be >= 0.");
+        
+        if (_parms._nparallelism == 0)
+            _parms._nparallelism = H2O.NUMCPUS;
+        
+        _predictorNames = extractPredNames(_dinfo, _dinfo._adaptedFrame.names().length-1);
+        _numPredictors = _predictorNames.length;
+        if (_parms._max_predictor_number < 1 || _parms._max_predictor_number > _numPredictors)
```

Comment:
It does.  However, it will take a long time for the algorithm to run.  Basically, you are asking the modelselection to build you the best 1-predictor model, best 2-predictors model, best 3-predictors model.... and final model with all predictors included.

Reply:
It does. However, it will take a long time for the algorithm to run. Basically, you are asking the modelselection to build you the best 1-predictor model, best 2-predictors model, best 3-predictors model.... and final model with all predictors included.

Reply:
Oh I see. It is ok then. :)